include/eve/traits/overload/supports.hpp

```diff
@@ -0,0 +1,268 @@
+// Overload's helpers
+//======================================================================================================================
+namespace eve::detail
+{
+  //====================================================================================================================
+  //  Delay the access to defaults() to after the proper operator() has been selected.
+  //  Also returns an empty set of options by default
+  //====================================================================================================================
+  template<typename T, typename...> EVE_FORCEINLINE constexpr auto defaults()
+  {
+    if constexpr(requires { T::defaults(); }) return T::defaults();
+    else                                      return options{};
+  }
+
+  //====================================================================================================================
+  // Internal concept checking if an option is supported by a specification
+  //====================================================================================================================
+  template<typename Spec, typename O>
+  concept supports_option = requires(Spec sp, O const& o) { { sp[o] }; };
+
+  //====================================================================================================================
+  // Internal concept checking if an option is supported by a any specification in a given subset
+  //====================================================================================================================
+  template<typename O, typename... Specs>
+  concept is_supported = (supports_option<Specs,O> || ... );
+
+  //====================================================================================================================
+  // Internal option carrying conditional mask or conditional expressions
+  //====================================================================================================================
+  struct condition_ : rbr::as_keyword<condition_>
+  {
+    using rbr::as_keyword<condition_>::operator=;
+    template<typename T> static constexpr bool check()
+    {
+      return conditional_expr<std::remove_cvref_t<T>> || logical_value<std::remove_cvref_t<T>>;
+    }
+  };
+}
+
+namespace eve
+{
+  //====================================================================================================================
+  //! @addtogroup extensions
+  //! @{
+  //!   @var condition
+  //!   @brief Keyword for retrieving conditionals decorator
+  //!
+  //!   When a eve::decorated_callable is called with an option provided via eve::conditional, the value of the
+  //!   conditional can be retrieved using eve::condition
+  //!   @see eve::supports
+  //! @}
+  //====================================================================================================================
+  inline constexpr detail::condition_ condition = {};
```

Comment:
Let's add smth like the word `key` or `option` to it, too many things want to be `condition`

Reply:
Will do